--- contrib/mem.lisp	2010-08-08 21:51:44.266000012 +0200
+++ contrib/mem.lisp	2010-08-09 17:04:19.530000007 +0200
@@ -1,6 +1,6 @@
 ;;; MEM formatters for the mode-line
 ;;;
-;;; Copyright 2009 Vitaly Mayatskikh
+;;; Copyright 2009 Vitaly Mayatskikh hacked by Bronsa
 ;;;
 ;;; Maintainer:
 ;;;
@@ -40,14 +40,7 @@
 (in-package :stumpwm.contrib.mem)
 
 ;; Install formatters.
-(dolist (a '((#\M fmt-mem-usage)
-             (#\N fmt-mem-usage-bar)))
-  (pushnew a *screen-mode-line-formatters* :test 'equal))
-
-;; Defaults arguments for fmt-mem-usage-bar
-(defvar *mem-usage-bar-width* 10)
-(defvar *mem-usage-bar-full* #\#)
-(defvar *mem-usage-bar-empty* #\:)
+(pushnew '(#\R fmt-mem-usage) *screen-mode-line-formatters* :test 'equal)
 
 (defun get-proc-fd-field (s field)
   (if s
@@ -69,18 +62,9 @@
 	   (read-from-string (get-proc-fd-field file "Buffers"))
 	   (read-from-string (get-proc-fd-field file "Cached"))))
       (setq allocated (- mem-total (+ mem-free buffers cached)))
-      (list mem-total allocated (/ allocated mem-total)))))
+      (/ allocated mem-total))))
 
 (defun fmt-mem-usage (ml)
   "Returns a string representing the current percent of used memory."
   (declare (ignore ml))
-  (let* ((mem (mem-usage))
-	 (|%| (truncate (* 100 (nth 2 mem))))
-	 (allocated (truncate (/ (nth 1 mem) 1000))))
-    (format nil "MEM: ~4D mb ^[~A~3D%^] " allocated (bar-zone-color |%|) |%|)))
-
-(defun fmt-mem-usage-bar (ml &optional (width *mem-usage-bar-width*) (full *mem-usage-bar-full*) (empty *mem-usage-bar-empty*))
-  "Returns a coloured bar-graph representing the current allocation of memory."
-  (declare (ignore ml))
-  (let ((cpu (truncate (* 100 (nth 2 (mem-usage))))))
-    (stumpwm::bar cpu width full empty)))
+  (format nil "~D%"(truncate (* 100 (mem-usage))))) 
