--- events.lisp	2010-08-09 22:57:00.553000060 +0200
+++ events.lisp	2010-08-09 22:58:02.069000060 +0200
@@ -127,7 +127,7 @@
                  (progn
                    (scale-screen screen new-heads)
                    (mapc 'group-add-head (screen-groups screen))
-                   (update-mode-lines screen))
+                   );(update-mode-lines screen))
                  (dformat 1 "Invalid configuration! ~S~%" new-heads)))))))))
 
 (define-stump-event-handler :map-request (parent send-event-p window)
@@ -341,9 +341,9 @@
 (defun update-window-properties (window atom)
   (case atom
     (:wm_name
-     (setf (window-title window) (xwin-name (window-xwin window)))
+     (setf (window-title window) (xwin-name (window-xwin window))))
      ;; Let the mode line know about the new name.
-     (update-all-mode-lines))
+     ;(update-all-mode-lines))
     (:wm_normal_hints
      (setf (window-normal-hints window) (get-normalized-normal-hints (window-xwin window))
            (window-type window) (xwin-type (window-xwin window)))
@@ -583,8 +583,8 @@
   (when (and window (eq mode :normal) (eq *mouse-focus-policy* :sloppy))
     (let ((win (find-window window)))
       (when (and win (find win (top-windows)))
-        (focus-all win)
-        (update-all-mode-lines)))))
+        (focus-all win)))))
+        ;(update-all-mode-lines)))))
 
 (define-stump-event-handler :button-press (window code x y child time)
   ;; Pass click to client
