--- floating-group.lisp	2010-05-30 22:03:17.625525819 +0200
+++ floating-group.lisp.new	2010-05-30 22:02:39.814650945 +0200
@@ -11,8 +11,8 @@
    (last-y :initform 0 :accessor float-window-last-y))
   )
 
-(defvar *float-window-border* 1)
-(defvar *float-window-title-height* 10)
+(defvar *float-window-border* 2)
+(defvar *float-window-title-height* 8)
 
 ;; some book keeping functions
 (defmethod (setf window-x) :before (val (window float-window))
@@ -57,7 +57,7 @@
     (setf (xlib:window-background (window-parent window))
           (xlib:alloc-color (xlib:screen-default-colormap (screen-number (window-screen window)))
                             (if (eq (group-current-window group) window)
-                                "Orange"
+                                "Grey20"
                                 "SteelBlue4")))
     (xlib:clear-area (window-parent window))))
 
